modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/properties/RegExPropertiesCollector.java

```diff
@@ -78,6 +79,36 @@ public RegExPropertiesCollector collect(File file) {
         return this;
     }
 
+    @Override
+    public RegExPropertiesCollector collect(URL url) {
```

Comment:
I didn't modified the FSDateExtractor since that one was specific for files on disk indeed. 
Updating it to extract the last-modified from the header.
The GridCoverage based are unrelated to the underlying source so I think they should work :-)